MobileWallet/Screens/Profile/ProfileViewController.swift

```diff
@@ -0,0 +1,155 @@
+    @IBOutlet weak var emojiLabel: UILabel!
+    @IBOutlet weak var copyEmojiButton: UIButton!
+    @IBOutlet weak var separatorView: UIView!
+    @IBOutlet weak var middleLabel: UILabel!
+    @IBOutlet weak var qrContainer: UIView!
+    @IBOutlet weak var qrImageView: UIImageView!
+
+    // MARK: - Override functions
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        customizeViews()
+    }
+
+    // MARK: - Private functions
+    private func customizeTitleLabel() {
+        let shareYourEmojiString = NSLocalizedString("Share your Emoji ID to receive Tari", comment: "profile title label")
+        let attributedString = NSMutableAttributedString(string: shareYourEmojiString,
+                                                         attributes: [.font: Theme.shared.fonts.profileTitleLightLabel ?? UIFont.systemFont(ofSize: 12.0),
+                                                                      .foregroundColor: Theme.shared.colors.profileTitleTextColor ?? .black,
+                                                                      .kern: -0.33])
+        attributedString.addAttribute(.font,
+                                      value: Theme.shared.fonts.profileTitleRegularLabel ?? UIFont.systemFont(ofSize: 12),
+                                      range: NSRange(location: 11, length: 8))
+
+        titleLabel.attributedText = attributedString
+    }
+
+    private func customizeEmoji() {
+        if let pubKey = TariLib.shared.tariWallet?.publicKey.0 {
+            let (emojis, _) = pubKey.emojis
+
+            self.emojiLabel.textColor = Theme.shared.colors.creatingWalletEmojisSeparator
+            self.emojiLabel.text = String(emojis.enumerated().map { $0 > 0 && $0 % 4 == 0 ? ["|", $1] : [$1]}.joined())
+        }
+    }
+
+    private func customizeCopyMyEmojiButton() {
+        let titleButton = NSLocalizedString("Copy my emoji ID", comment: "Profile title button")
+        self.copyEmojiButton.setTitle(titleButton, for: .normal)
+        self.copyEmojiButton.titleLabel?.font = Theme.shared.fonts.profileCopyEmojiButton
+        self.copyEmojiButton.setTitleColor(Theme.shared.colors.profileCopyEmojiColor ?? .blue, for: .normal)
+    }
+
+    private func customizeSeparatorView() {
+        separatorView.backgroundColor = Theme.shared.colors.profileSeparatorView ?? .gray
+    }
+
+    private func customizeMiddleLabel() {
+        let middleLabelText = NSLocalizedString("Transacting in person? Your sender can get your emoji ID by scanning this QR code in their Tari app", comment: "Profile middle label")
+
+        self.middleLabel.text = middleLabelText
+        self.middleLabel.font = Theme.shared.fonts.profileMiddleLabel
+        self.middleLabel.textColor = Theme.shared.colors.profileMiddleLabel ?? .gray
+    }
+
+    private func genQRCode() {
+
+        guard let wallet = TariLib.shared.tariWallet else {
+            return
+        }
+
+        let (walletPublicKey, _) = wallet.publicKey
```

Comment:
Rather handle any potential errors returned from these lib functions. Can use `UserFeedback.shared.error` function to display a message to the user. We can later add proper logging and report on any errors the user might have seen instead of silently ignoring them.